CQSDIContainer/CQSDIContainer/CQSFactory.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Castle.MicroKernel;
+using Castle.Windsor;
+using CQSDIContainer.Attributes;
+using CQSDIContainer.CommandDecorators;
+using CQSDIContainer.Commands;
+using CQSDIContainer.QueryDecorators;
+using IQ.Platform.Framework.Common.CQS;
+
+namespace CQSDIContainer
+{
+	public interface ICQSFactory
+	{
+		ICommandHandler<TCommand> CreateCommandHandler<TCommand>() where TCommand : ICommand;
+		IQueryHandler<TQuery, TResult> CreateQueryHandler<TQuery, TResult>() where TQuery : IQuery<TResult>;
+	}
+
+	public class CQSFactory : ICQSFactory
+	{
+		private readonly IWindsorContainer _container;
+
+		public CQSFactory(IWindsorContainer container)
```

Comment:
I'd do something different for RQ than what is done for instantiation here.  Not sure what yet.